utopia-vscode-extension/src/extension.ts

```diff
@@ -334,56 +307,106 @@ function getFullConfig(): UtopiaVSCodeConfig {
 let currentDecorations: Array<DecorationRange> = []
 let currentSelection: BoundsInFile | null = null
 
-function initMessaging(context: vscode.ExtensionContext, workspaceRootUri: vscode.Uri): void {
-  function handleMessage(message: ToVSCodeMessage): void {
-    switch (message.type) {
-      case 'OPEN_FILE':
-        if (message.bounds != null) {
-          revealRangeIfPossible(workspaceRootUri, { ...message.bounds, filePath: message.filePath })
-        } else {
-          openFile(vscode.Uri.joinPath(workspaceRootUri, message.filePath))
-        }
-        break
-      case 'UPDATE_DECORATIONS':
-        currentDecorations = message.decorations
-        updateDecorations(currentDecorations)
-        break
-      case 'SELECTED_ELEMENT_CHANGED':
-        const followSelectionEnabled = getFollowSelectionEnabledConfig()
-        const shouldFollowSelection =
-          followSelectionEnabled &&
-          (shouldFollowSelectionWithActiveFile() || message.forceNavigation === 'force-navigation')
-        if (shouldFollowSelection) {
-          currentSelection = message.boundsInFile
-          revealRangeIfPossible(workspaceRootUri, message.boundsInFile)
-        }
-        break
-      case 'GET_UTOPIA_VSCODE_CONFIG':
-        sendFullConfigToUtopia()
-        break
-      case 'SET_FOLLOW_SELECTION_CONFIG':
-        vscode.workspace
-          .getConfiguration()
-          .update(FollowSelectionConfigKey, message.enabled, vscode.ConfigurationTarget.Workspace)
-        break
-      case 'SET_VSCODE_THEME':
-        vscode.workspace.getConfiguration().update('workbench.colorTheme', message.theme, true)
-        break
-      case 'ACCUMULATED_TO_VSCODE_MESSAGE':
-        for (const innerMessage of message.messages) {
-          handleMessage(innerMessage)
-        }
-        break
-      case 'UTOPIA_READY':
-        sendMessage(vsCodeReady())
-        break
-      default:
-        const _exhaustiveCheck: never = message
-        console.error(`Unhandled message type ${JSON.stringify(message)}`)
-    }
-  }
+function sendMessageToUtopia(message: FromVSCodeToUtopiaMessage): void {
+  vscode.commands.executeCommand('utopia.toUtopiaMessage', message)
+}
```

Comment:
This part is the crux of the new communication approach. We have this extension and VS Code register a pair of new commands for passing messages between themselves, with the VS Code side then forwarding those messages straight onto Utopia via a `window.postMessage` (since that isn't possible from the extension).